plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANATable.java

```diff
@@ -47,4 +57,41 @@ public List<? extends HANATableColumn> getAttributes(@NotNull DBRProgressMonitor
     public HANATableColumn getAttribute(@NotNull DBRProgressMonitor monitor, @NotNull String attributeName) throws DBException {
         return (HANATableColumn) super.getAttribute(monitor, attributeName);
     }
+
+    @Override
+    public boolean hasStatistics() {
+        return tableSize != -1;
+    }
+
+    @Override
+    public long getStatObjectSize() {
+        return tableSize;
+    }
+
+    void fetchStatistics(JDBCResultSet dbResult) throws SQLException {
+        tableSize = dbResult.getLong("DISK_SIZE");
+    }
```

Comment:
Probably better to make it public then

Reply:
What for?

Reply:
Well, just prefer to avoid package private methods, but ok.